Formula/cppcheck.rb

```diff
@@ -60,7 +82,72 @@ def install
         cout << "Hello World!" << endl;
         return 0;
       }
+
+      class Example
+      {
+        public:
+          int GetNumber() const;
+          explicit Example(int initialNumber);
+        private:
+          int number;
+      };
+
+      Example::Example(int initialNumber)
+      {
+        number = initialNumber;
+      }
+    EOS
+    system "#{bin}/cppcheck", test_cpp_file
+
+    # Test the "out of bounds" check
+    test_cpp_file_check = testpath/"testcheck.cpp"
+    test_cpp_file_check.write <<-EOS.undent
+      int main()
+      {
+      char a[10];
+      a[10] = 0;
+      return 0;
+      }
     EOS
-    system "#{bin}/cppcheck", "test.cpp"
+    output = shell_output("#{bin}/cppcheck #{test_cpp_file_check} 2>&1")
+    assert_match "out of bounds", output, "Expected #{name} to detect out of bounds"
```

Comment:
You can omit the third argument here and in the following asserts. We usually don't care about the message and the reason for the failure, if it happens, is obvious from the code. It also makes the code shorter. :smiley_cat: